File: app/src/main/java/com/adonai/manman/entities/ManSectionItem.java
package com.adonai.manman.entities;

import android.support.annotation.NonNull;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

/**
 * Object represents an item in browse-chapter page of mankier.com
 *
 * It also serves as DB entity-mirroring class
 */
@DatabaseTable(tableName = "man_chapters")
public class ManSectionItem implements Comparable<ManSectionItem> {

    // these are filled by page
    @DatabaseField(id = true)
    private String url;

    @DatabaseField(canBeNull = false, index = true)
    private String name;

    @DatabaseField
    private String description;

    // this is filled with constant parent chapter from resources
    @DatabaseField(index = true, canBeNull = false)
    private String parentChapter;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getParentChapter() {
        return parentChapter;
    }

    public void setParentChapter(String parentChapter) {
        this.parentChapter = parentChapter;
    }

    @Override
    public int compareTo(@NonNull ManSectionItem another) {
        return Character.compare(name.charAt(0), another.name.charAt(0));
    }
}

File: app/src/main/java/com/adonai/manman/entities/ManSectionItem.kt
package com.adonai.manman.entities

import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable

/**
 * Object represents an item in browse-chapter page of mankier.com
 * It also serves as DB entity-mirroring class
 *
 * @author Kanedias
 */
@DatabaseTable(tableName = "man_chapters")
class ManSectionItem : Comparable<ManSectionItem> {

    // these are filled by page
    @DatabaseField(id = true)
    lateinit var url: String

    @DatabaseField(canBeNull = false, index = true)
    lateinit var name: String

    @DatabaseField(canBeNull = false)
    lateinit var description: String

    // this is filled with constant parent chapter from resources
    @DatabaseField(index = true, canBeNull = false)
    lateinit var parentChapter: String

    override fun compareTo(other: ManSectionItem): Int {
        return name.compareTo(other.name)
    }
}